app/models/order.rb

```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
 class Order < ApplicationRecord
+  serialize :reward_snapshot
+
   belongs_to :employee
   belongs_to :reward
+
+  def snapshot_price
```

Comment:
I think this name might be misleading -  I would rather use something like `transaction_price`, or simply `price`